components/core/routes/Acl.py

```diff
@@ -0,0 +1,47 @@
+from flask import json
+from flask import request, abort, Response, g
+from utils.token_utils import token_validator
+from managers.AclManager import *
+
+
+def check_access(id, user_id):
+    token = token_validator(request.headers.get('token'))
+    if token is None:
+        return '{"error":"token error"}', 403
+    try:
+        entity = request.args.get('type')
+        if not entity == 'fr' and not entity =='fl':
+            return Response('{"error":"invalid parameter"}', status=400)
+
+        check_response = get_access(id, user_id, entity)
+        if not isinstance(check_response, str):
+            resp = Response(response=json.dumps(check_response), status=200)
+        else:
+            resp = Response('{"error":"' + check_response + '"}', status=500)
+    except Exception as e:
+        resp = Response('{"error":"' + str(e) + '"}', status=400)
+    resp.headers['token'] = token
+    resp.headers['Access-Control-Expose-Headers'] = 'token'
+    return resp
+
+
+def grant_access(id):
+    token = token_validator(request.headers.get('token'))
+    if token is None:
+        return '{"error":"token error"}', 403
+    try:
+        entity = request.args.get('type')
+        if not entity == 'fr' and not entity =='fl':
+            return Response('{"error":"invalid parameter"}', status=400)
+
+        user_name = request.form['user_name']
+        access = request.form['access']
+
+        grant_response = give_access(id, user_name, entity, access)
+        resp = Response(response='{"status":"' + grant_response + '"}',
```

Comment:
I guess 500 is not a suitable code for permission error, 4xx? @VibhorCodecianGupta

Reply:
this is not a permission error, this is made in a way that 4xx errors will be either token error 403 (forbidden), a 400 (bad request) code if the query param is invalid or a 400 when some exception is thrown otherwise. This line gets the response from the function that hits the db, so it's either 200 or 500 server error if something goes wrong. the 4xx errors are all handled prior to this line. thoughts?

Reply:
If the user doesn't have access, it doesn' mean `error` I think. 403 (forbidden) shows you can't perform this action. 
Let's say a user has correct token but it doesn't belong to the user who is trying to perform this action then it's a Forbidden state I think 
@agentmilindu @JaDogg what do you think?

Reply:
That's totally correct. I'm talking about the 500 code you commented on, I mean that because all the 4xx are handled before this 500 code, and because the code reaches here only if the user inputs are correct and the user has access, this method has already bypassed the 4xx errors. the only errors possible here are server errors if the method doesn't work, for example, db connection times out. which is a 500 error

Reply:
This is what I use/recommend:
* 403 - any permission issue. Forbidden access - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
* 422 - input is parsed (read as JSON) but it's wrong - ex: fields are wrong, invalid data type, invalid enum (fr/fl in this case) https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
* 500 - Avoid responding with a 500 intentionally. This is reserved for situations where there is some error on server side. (Unexpected exception, etc.)  https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500